src/tools/perf/lib/libperf_memory.c

```diff
@@ -168,43 +169,80 @@ void ucp_perf_test_free_mem(ucx_perf_context_t *perf)
     ucp_perf_mem_free(perf, perf->ucp.send_memh);
 }
 
-static void
-ucx_perf_test_memcpy_host(void *dst, ucs_memory_type_t dst_mem_type,
-                          const void *src, ucs_memory_type_t src_mem_type,
-                          size_t count)
+ucs_status_t uct_perf_test_alloc_mem_by_mem_type(ucx_perf_context_t *perf,
```

Comment:
The problem was `uct_iface_mem_alloc` is using `UCS_MEMORY_TYPE_HOST` hard-coded, maybe we can extract the logic into another function which will accept memory type, in some inner API `.h` file 
![image](https://github.com/openucx/ucx/assets/19361875/785ec082-cf7a-4ca0-86ec-f0235f861006)

Reply:
IMO extracting shared logic to a third func, which will be called both by `uct_iface_mem_alloc ` and `uct_perf_test_alloc_mem_by_mem_type` is justified, in order to avoid this big code duplication. @yosefe  WDYT?

Reply:
But this function will have to be in UCT API. And because of backward compatibility, it will have params struct. So IMO additional function will not improve the situation much.
Can't we just pass 2 allocations methods to uct_mem_alloc and let it do the md_query logic internally?

Reply:
Your suggestion would require changng the way `uct_mem_alloc` works..?

Reply:
if uct_mem_alloc is not supporting what we need, we can extend it. what is missing?
btw @roiedanino is this PR needed only for UCT level perftest? if yes we should put it on hold as discussed

Reply:
It is needed for both UCT and UCP perftest, UCP is still using allocators in the initialization

Reply:
it doesn't make sense - we should be able to test UCP level performance without using UCT api for memory allocation.
Should use ucp_mem_map for this
Maybe this PR can be refactored to keep only UCP support for now?

Reply:
I can try, I'll open a new branch with a new PR if that's ok

Reply:
yes, thanks!